Hunting Queries/Box/BoxSuspiciousFiles.yaml

```diff
@@ -0,0 +1,22 @@
+id: 6b91dda7-d9c5-4197-9dea-0c41f7c55176
+name: Box - Suspicious or sensitive files
+description: |
+  'Query searches for potentially suspicious files or files which can contain sensitive information such as passwords, secrets.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: BoxDataConnector
+    dataTypes:
+      - BoxEvents_CL
+tactics:
+  - PrivilegeEscalation
+query: |
+  BoxEvents
+  | where TimeGenerated > ago(24h)
+  | where SourceItemName == 'id_rsa' or SourceItemName contains 'password' or SourceItemName contains 'key' or SourceItemName contains '_key' or SourceItemName contains '.ssh' or SourceItemName endswith '.npmrc' or SourceItemName endswith '.muttrc' or SourceItemName contains 'config.json' or SourceItemName contains '.gitconfig' or SourceItemName endswith '.netrc' or SourceItemName endswith 'package.json' or SourceItemName endswith 'Gemfile' or SourceItemName endswith 'bower.json' or SourceItemName endswith 'config.gypi' or SourceItemName endswith 'travis.yml' or SourceItemName endswith '.ps1' or SourceItemName endswith '.bat' or SourceItemName endswith '.scr' or SourceItemName endswith '.sh' or SourceItemName endswith '.exe' or SourceFileName == 'id_rsa' or SourceFileName contains 'password' or SourceFileName contains 'key' or SourceFileName contains '_key' or SourceFileName contains '.ssh' or SourceFileName endswith '.npmrc' or SourceFileName endswith '.muttrc' or SourceFileName contains 'config.json' or SourceFileName contains '.gitconfig' or SourceFileName endswith '.netrc' or SourceFileName endswith 'package.json' or SourceFileName endswith 'Gemfile' or SourceFileName contains 'bower.json' or SourceFileName contains 'config.gypi' or SourceFileName contains 'travis.yml' or SourceFileName endswith '.ps1' or SourceFileName endswith '.bat' or SourceFileName endswith '.scr' or SourceFileName endswith '.sh' or SourceFileName endswith '.exe'

```

Comment:
contains is not recommended, try using `has_any` - https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatypes-string-operators

Reply:
updated

Reply:
does not look like this was updated

Reply:
has_any is not applicable in this case. If we use has_any the rule will not cover all desired events(parts of file name)